src/parquet/encodings/encoding-test.cc

```diff
@@ -301,7 +300,7 @@ TYPED_TEST(TestDictionaryEncoding, BasicRoundTrip) {
 
 TEST(TestDictionaryEncoding, CannotDictDecodeBoolean) {
   PlainDecoder<Type::BOOLEAN> dict_decoder(nullptr);
-  DictionaryDecoder<Type::BOOLEAN> decoder(nullptr);
+  DictionaryDecoder<Type::BOOLEAN> decoder(nullptr, get_default_pool());
```

Comment:
Per above I don't think it makes sense to sprinkle `get_default_pool` everywhere — it adds complexity where not strictly necessary.